frontend/app/src/store/tasks/index.ts

```diff
@@ -57,7 +59,11 @@ export const useTaskStore = defineStore('tasks', () => {
 
   const registerHandler = <R, M extends TaskMeta>(
     task: TaskType,
-    handlerImpl: (actionResult: ActionResult<R>, meta: M) => void,
+    handlerImpl: (
+      actionResult: ActionResult<R>,
+      meta: M,
+      error?: Error
```

Comment:
Any way to use an enriched `ActionResult` internally that also includes the error? instead of passing it as a parameter?